### ISSUE TITLE
[Extensibility Bug] pageextension Caption property translate

### ISSUE THREAD

**Issue by RadoArvay** [microsoft/ALAppExtensions#14996]
I am not sure if this is the right place to enter this issue but it seems that the translation of Caption property for a pageextension object does not work.

Example:
```al
pageextension 51612509 "slt_VAT Report List" extends "VAT Report List"
{
    Caption = 'VAT Returns (some text)';
    ...
}
```
When I publish the app and I use 'Tell me' (search), I can see modified title of the page in the list. But when I change the language, the translation from BaseApp is used instead of the translation provided by the app. Both translations (BaseApp and the app) are our own - for SK country.

**Comment by dnpb** [comment]
If it's what I think you mean, I went through a support process with MS to get an answer on this. The full detail is posted here:
https://github.com/microsoft/AL/issues/6609

Here is MS's summary, followed by screengrabs of their PowerPoint presentation, with tables of the 'weighting'/precedence of different sources of translations, with numerically higher ones winning out.

***

We have the possibility to share with you a little test scenario with Caption and Tooltip and also explaining how the translation works.

Attached you will find the slide presentation we used in our teams meeting.

We explained both scenarios highlighted on this ticket.

Let’s share our explanation for the ToolTip.

When you are using, what we call, a label property like Caption or Tooltip, then the language it is going to target is English-United States (ENU), which means that

    Caption = ‘My English Text’;

Is the same as

    CaptionML = ENU = ‘My English Text’;

You have a GB tenant which comes with English-Great Britain (ENG) translations.

So when the system has to pick a translation it tries to find the best match, and in the mix of possible translations it finds

* ENU = ‘Test’
* ENG = ‘Specifies the ….’
* And maybe more.

So when you set the language in settings to ENG, then it starts out by looking at can it find a translation with ENG, and it does so the text ‘Specifies the…’ is shown.

When you then change to ENU then it also get’s a match on the user specified language and text ‘Test’ is shown..

Now the explanation for the Caption.

When it comes to finding the translation for Page Controls then it uses the same logic as for Page Actions, with a small twist. If it can’t find any translation that matches the language the user has set, including the fallback to the primary language of the user specified one, and ENU (ENU is hard coded) then it looks at the table field that is referenced in the page control.

The control called ‘Name’ doesn’t have a caption on it, but all table fields does, so the table field has a translation for ENG.
When you then write a caption on the page control, then it is getting caught by the fallback logic, before it starts to look at the table fields.

And last for the Caption on the TableExtensions modify field, is the same story as for the ToolTip.

All in all, the system works as intended. It might not be the most intuitive system as we shared, but we are working with docs writer about making a page about how all of this works, so all our Customers and Partners better understand it._

***

![image](https://user-images.githubusercontent.com/63455896/118833313-428e1380-b8b9-11eb-90cf-1912d3b7da1a.png)

![image](https://user-images.githubusercontent.com/63455896/118833380-4cb01200-b8b9-11eb-9821-b6747d068c08.png)

![image](https://user-images.githubusercontent.com/63455896/118833414-546fb680-b8b9-11eb-8487-cd1c736ec3c9.png)
